src/input/plugins/CdioParanoiaInputPlugin.cxx

```diff
@@ -251,14 +254,25 @@ input_cdio_open(const char *uri,
 				      be);
 	}
 
-	lsn_t lsn_from, lsn_to;
-	if (parsed_uri.track >= 0) {
-		lsn_from = cdio_get_track_lsn(cdio, parsed_uri.track);
-		lsn_to = cdio_get_track_last_lsn(cdio, parsed_uri.track);
-	} else {
-		lsn_from = 0;
-		lsn_to = cdio_get_disc_last_lsn(cdio);
-	}
+	lsn_t lsn_to, lsn_from;
+
+	if(parsed_uri.track < 0){
+		lsn_from = cdio_cddap_disc_firstsector(drv);
+		lsn_to = cdio_cddap_disc_lastsector(drv);
+	} else{
+		lsn_from = cdio_cddap_track_firstsector(drv, parsed_uri.track);
+		lsn_to = cdio_cddap_track_lastsector(drv, parsed_uri.track);
+	} 
+
+	/* LSNs < 0 indicate errors (e.g. -401: Invaid track, -402: no pregap) */
+	if(lsn_from < 0 || lsn_to < 0)
+		throw FmtRuntimeError("Error {} on track {}",
+							lsn_from < 0 ? lsn_from : lsn_to, parsed_uri.track);
```

Comment:
If you find a mistake like this wrong indent in your commit before your PR gets merged, please amend the commit instead of adding a fixup commit.
I tried to review your comment, found this bad indentation, and thought about adding a comment; but then I saw you had a fixup. If you avoid this noise by amending known-bad commits, you save me some confusion.